### PR TITLE
CIV-0000 fix sprint p135 4 cosc

### PR DESCRIPTION
**Pull request by LeonardoPalmeiro**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/common/models/generalApplication/GeneralApplication.ts [hunk 1]
```diff
@@ -53,7 +55,8 @@ export class GeneralApplication {
     wantToUploadDocuments?: YesNo,
     uploadN245Form?: UploadGAFiles,
     informOtherParties?: InformOtherParties,
-    applicationFee?: ClaimFeeData) {
+    applicationFee?: ClaimFeeData,
+    certificateOfSatisfactionOrCanceled?: CertificateOfSatisfactionOrCanceled) {
```
Comment (hmcts-version1-ignacio) [inline src/main/common/models/generalApplication/GeneralApplication.ts:59]: It should be "certificateOfSatisfactionOrCancellation"

File: src/main/modules/i18n/locales/cy.json [hunk 2]
```diff
@@ -2726,7 +2727,6 @@
       },
       "DEBT_PAYMENT": {
         "DO_YOU_HAVE_EVIDENCE": "Do you have evidence of the debt payment?",
-        "YOU_WILL_NEED_EVIDENCE": "You’ll need to provide evidence that you’ve paid the debt.",
```
Comment (sherlynkhaw) [inline src/main/modules/i18n/locales/cy.json:2729]: image
please don't delete this line, we need it

Comment (sherlynkhaw) [reply src/main/modules/i18n/locales/cy.json:2729]: delete line 2742>

Comment (LeonardoPalmeiro) [reply src/main/modules/i18n/locales/cy.json:2729]: no, you don't need :) I changed to COMMON.ASK_FOR_PROOF_OF_DEBT_PAYMENT

File: src/main/common/models/generalApplication/GeneralApplication.ts [hunk 3]
```diff
@@ -53,7 +55,8 @@ export class GeneralApplication {
     wantToUploadDocuments?: YesNo,
     uploadN245Form?: UploadGAFiles,
     informOtherParties?: InformOtherParties,
-    applicationFee?: ClaimFeeData) {
+    applicationFee?: ClaimFeeData,
+    certificateOfSatisfactionOrCanceled?: CertificateOfSatisfactionOrCancellation) {
```
Comment (hmcts-version1-ignacio) [inline src/main/common/models/generalApplication/GeneralApplication.ts:59]: "certificateOfSatisfactionOr**Cancellation**?: CertificateOfSatisfactionOrCancellation"

Comment (LeonardoPalmeiro) [reply src/main/common/models/generalApplication/GeneralApplication.ts:59]: I changed to certificateOfSatisfactionOrCancellation as your sugestion

File: src/main/modules/i18n/locales/cy.json (hunk 2 above)
Comment (sherlynkhaw) [reply src/main/modules/i18n/locales/cy.json:2729]: "YOU_WILL_NEED_EVIDENCE": "You’ll need to provide evidence that you’ve paid the debt.",
"ASK_FOR_PROOF_OF_DEBT_PAYMENT": "Confirm you've paid a judgment debt",

these two lines are different. you're deleting the wrong line

File: src/main/views/features/generalApplication/certOfSorC/debt-payment-evidence.njk [hunk 4]
```diff
@@ -15,7 +15,7 @@
           <span class="govuk-caption-l">{{ t('PAGES.GENERAL_APPLICATION.DEBT_PAYMENT.CONFIRM_PAID') }}</span>
           <h1 class="govuk-heading-l">{{ t('PAGES.GENERAL_APPLICATION.DEBT_PAYMENT.DO_YOU_HAVE_EVIDENCE') }}</h1>
           <p class="govuk-body">
-            {{ t('PAGES.GENERAL_APPLICATION.DEBT_PAYMENT.YOU_WILL_NEED_EVIDENCE')}}
+            {{ t('COMMON.ASK_FOR_PROOF_OF_DEBT_PAYMENT')}}
```
Comment (sherlynkhaw) [inline src/main/views/features/generalApplication/certOfSorC/debt-payment-evidence.njk:18]: "YOU_WILL_NEED_EVIDENCE": "You’ll need to provide evidence that you’ve paid the debt.",
"ASK_FOR_PROOF_OF_DEBT_PAYMENT": "Confirm you've paid a judgment debt",

these two lines are different. you're deleting the wrong line

Comment (sherlynkhaw) [reply src/main/views/features/generalApplication/certOfSorC/debt-payment-evidence.njk:18]: will fix this now

File: src/main/routes/features/generalApplication/certOfSorC/defendantFinalPaymentDateController.ts [hunk 5]
```diff
@@ -2,28 +2,30 @@ import {NextFunction, Request, Response, Router} from 'express';
...
 } from 'routes/urls';
-import {
-  defendantFinalPaymentDateService,
-} from 'services/features/generalApplication/certOfSorC/defendantFinalPaymentDateService';
 import {GenericForm} from 'form/models/genericForm';
 import {constructResponseUrlWithIdParams} from 'common/utils/urlFormatter';
 import {getCancelUrl} from 'services/features/generalApplication/generalApplicationService';
+import {CertificateOfSatisfactionOrCancellation} from 'models/generalApplication/CertificateOfSatisfactionOrCancellation';
+import {
+  getCertificateOfSatisfactionOrCanceled, saveCertificateOfSatisfactionOrCanceled,
```
Comment (hmcts-version1-ignacio) [inline src/main/routes/features/generalApplication/certOfSorC/defendantFinalPaymentDateController.ts:13]: getCertificateOfSatisfactionOr**Cancellation**

File: src/main/routes/features/generalApplication/certOfSorC/debtPaymentEvidenceController.ts [hunk 6]
```diff
@@ -1,17 +1,18 @@
...
 import {getCancelUrl} from 'services/features/generalApplication/generalApplicationService';
 import {debtPaymentOptions} from 'models/generalApplication/debtPaymentOptions';
 import {constructResponseUrlWithIdParams} from 'common/utils/urlFormatter';
 import {GenericForm} from 'form/models/genericForm';
 import {DebtPaymentEvidence} from 'models/generalApplication/debtPaymentEvidence';
-import {debtPaymentEvidenceService} from 'services/features/generalApplication/certOfSorC/debtPaymentEvidenceService';
-import {generateRedisKey} from 'modules/draft-store/draftStoreService';
-import {AppRequest} from 'models/AppRequest';
+import {
+  getCertificateOfSatisfactionOrCanceled,
```
Comment (hmcts-version1-ignacio) [inline src/main/routes/features/generalApplication/certOfSorC/debtPaymentEvidenceController.ts:12]: getCertificateOfSatisfactionOr**Cancellation**, saveCertificateOfSatisfactionOr**Cancellation**

File: src/main/routes/features/generalApplication/certOfSorC/debtPaymentEvidenceController.ts [hunk 7]
```diff
@@ -1,17 +1,18 @@
...
+  saveCertificateOfSatisfactionOrCanceled,
+} from 'services/features/generalApplication/certOfSorC/certificateOfSatisfactionOrCanceledService';
```
Comment (hmcts-version1-ignacio) [inline src/main/routes/features/generalApplication/certOfSorC/debtPaymentEvidenceController.ts:14]: Here too.

File: src/main/routes/features/generalApplication/certOfSorC/defendantFinalPaymentDateController.ts [hunk 8]
```diff
@@ -2,28 +2,30 @@ import {NextFunction, Request, Response, Router} from 'express';
...
+const title = 'COMMON.ASK_FOR_PROOF_OF_DEBT_PAYMENT';
 
 defendantPaymentDateController
   .get(
     COSC_FINAL_PAYMENT_DATE_URL, async (req: Request, res: Response, next: NextFunction) => {
       try {
         const cancelUrl = await getCancelUrl(req.params.id, null);
         const backLinkUrl = constructResponseUrlWithIdParams(req.params.id, GA_ASK_PROOF_OF_DEBT_PAYMENT_GUIDANCE_URL);
-        const defendantFinalPaymentDate = await defendantFinalPaymentDateService.getDefendantResponse(req);
+        const certificateOfSatisfactionOrCanceled: CertificateOfSatisfactionOrCancellation = await getCertificateOfSatisfactionOrCanceled(req);
```
Comment (hmcts-version1-ignacio) [inline src/main/routes/features/generalApplication/certOfSorC/defendantFinalPaymentDateController.ts:27]: certificateOfSatisfactionOr**Cancellation**

File: src/main/routes/features/generalApplication/uploadEvidenceDocumentForApplication.ts [hunk 9]
```diff
@@ -37,7 +37,7 @@ async function renderView(form: GenericForm<UploadGAFiles>, claim: Claim, claimI
   const cancelUrl = await getCancelUrl(claimId, claim);
   const currentUrl = constructResponseUrlWithIdParams(claimId, GA_UPLOAD_DOCUMENTS_URL);
   const isConfirmPaidCCJAppType = isConfirmYouPaidCCJAppType(claim);
-  const backLinkPage = isConfirmPaidCCJAppType ? GA_DEBT_PAYMENT_EVIDENCE_URL : GA_WANT_TO_UPLOAD_DOCUMENTS_URL;
+  const backLinkPage = isConfirmPaidCCJAppType ? GA_DEBT_PAYMENT_EVIDENCE_COSC_URL : GA_WANT_TO_UPLOAD_DOCUMENTS_URL;
   const backLinkUrl = constructResponseUrlWithIdParams(claimId, backLinkPage);
   const headerTitle = isConfirmPaidCCJAppType ? 'PAGES.GENERAL_APPLICATION.SELECTED_APPLICATION_TYPE.CONFIRM_YOU_PAID_CCJ' : getDynamicHeaderForMultipleApplications(claim);
```
Comment (kalachandrasekar1) [inline src/main/routes/features/generalApplication/uploadEvidenceDocumentForApplication.ts:42]: headerTitle also needs to update with common field

File: src/main/services/features/generalApplication/certOfSorC/certificateOfSatisfactionOrCanceledService.ts
Comment (hmcts-version1-ignacio) [inline src/main/services/features/generalApplication/certOfSorC/certificateOfSatisfactionOrCanceledService.ts:1]: src/main/services/features/generalApplication/certOfSorC/certificateOfSatisfactionOrCancellationService.ts

File: src/main/routes/features/generalApplication/certOfSorC/defendantFinalPaymentDateController.ts (hunk 5 above)
Comment (sherlynkhaw) [reply src/main/routes/features/generalApplication/certOfSorC/defendantFinalPaymentDateController.ts]: fixed all but best if you double check!

File: src/test/unit/routes/features/claim/generalApplication/certOfSorC/askProofOfDebtPaymentGuidanceController.test.ts [hunk 10]
```diff
@@ -47,8 +47,8 @@ describe('General Application - ask proof of debt payment guidance', () => {
 
           const decodedText = res.text.replace(/&#39;/g, "'");
 
-          expect(decodedText).toContain(t('PAGES.GENERAL_APPLICATION.ASK_FOR_PROOF_OF_DEBT_PAYMENT.PAGE_TITLE'));
-          expect(decodedText).toContain(t('PAGES.GENERAL_APPLICATION.ASK_FOR_PROOF_OF_DEBT_PAYMENT.TITLE'));
+          expect(decodedText).toContain(t('COMMON.ASK_FOR_PROOF_OF_DEBT_PAYMENT'));
```
Comment (hmcts-version1-ignacio) [inline src/test/unit/routes/features/claim/generalApplication/certOfSorC/askProofOfDebtPaymentGuidanceController.test.ts:50]: Extra check!

Comment (sherlynkhaw) [reply src/test/unit/routes/features/claim/generalApplication/certOfSorC/askProofOfDebtPaymentGuidanceController.test.ts:50]: remvoed!

File: src/main/routes/features/generalApplication/uploadEvidenceDocumentForApplication.ts (hunk 9 above)
Comment (sherlynkhaw) [reply src/main/routes/features/generalApplication/uploadEvidenceDocumentForApplication.ts]: fixed, could you please check that it's correct @kalachandrasekar1

**Review by kalachandrasekar1 — APPROVED** [review]

**Pull request merged**